packages/repository/src/__tests__/acceptance/has-one.relation.acceptance.ts

```diff
@@ -115,6 +115,70 @@ describe('hasOne relation', () => {
     ).to.be.rejectedWith(EntityNotFoundError);
   });
 
+  it('can PATCH hasOne instances', async () => {
+    existingCustomerId = 11;
+    const address = await controller.createCustomerAddress(existingCustomerId, {
+      street: '1 Amedee Bonnet',
+      zipcode: '69740',
+      city: 'Genas',
+      province: 'Rhone',
+    });
+
+    const persisted = await addressRepo.findById(address.zipcode);
+    expect(persisted.toObject()).to.deepEqual(address.toObject());
+
+    const patchObject = {city: 'Lyon-Genas'};
+    const arePached = await controller.patchCustomerAddress(
+      existingCustomerId,
+      patchObject,
+    );
+
+    expect(arePached.count).to.equal(1);
+    const patchedData = await addressRepo.findById(address.zipcode);
+    expect(patchedData.city).to.containEql(address.city);
+  });
+
+  it('can PUT hasOne instances', async () => {
+    existingCustomerId = 11;
+    const address = await controller.createCustomerAddress(existingCustomerId, {
+      street: '1 Amedee Bonnet',
+      zipcode: '69740',
+      city: 'Genas',
+      province: 'Rhone',
+    });
+
+    const persisted = await addressRepo.findById(address.zipcode);
+    expect(persisted.toObject()).to.deepEqual(address.toObject());
+
+    const putObject = {
+      city: 'Lyon-Genas',
+      province: 'Rhone-Alpe',
+      zipcode: '69740',
+    };
+    await controller.putCustomerAddress(existingCustomerId, putObject);
+    const putData = await addressRepo.findById(address.zipcode);
```

Comment:
I find the name `putData` little bit confusing. How about `replacedInstance`?

Reply:
As per your proposition, I removed from the PUT operation test; it will be implemented in a later phase.